lib/retrieveFileStreams.js

```diff
@@ -0,0 +1,13 @@
+function retrieveFileStreams (conn, fileStreamArray, client_type) {
+  return Promise.all(fileStreamArray.map((file) => {
+    // make sure we are using sftp library
+    if (client_type === "sftp") {
+      // make sure we are returning files and not directories
+      if (file.type === "-") {
+        return conn.get(file.name)
+      }
+    }
```

Comment:
If you're not using sftp or have a file type of "-" you should either return a rejected promise or a resolved promise, correct?

Also, how much logging do you want of those scenarios?